pilot/pkg/serviceregistry/kube/controller/ambient/services.go

```diff
@@ -352,15 +383,28 @@ func MakeSource(o controllers.Object) model.TypedObject {
 	}
 }
 
+// ServiceEntryInfo is a wrapper around ServiceInfo that handles key conflicts
+// on hostname.
+type ServiceEntryInfo struct {
+	model.ServiceInfo
+}
+
+func (s ServiceEntryInfo) ResourceName() string {
+	return s.GetNamespace() + "/" + s.GetName()
+}
```

Comment:
Would this cause a problem when a SE has 2 host names? We'd wind up with two ServiceEntryInfo which have the same ResourceName in that case even though internally their ServiceInfo are unique.